monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -102,39 +113,56 @@ def get_tunnels():
     def get_scanned():
         formatted_nodes = []
 
-        nodes = ReportService.get_all_displayed_nodes()
+        machines = ReportService.get_all_machines()
 
-        for node in nodes:
+        for machine in machines:
             # This information should be evident from the map, not sure a table/list is a good way
             # to display it anyways
-            nodes_that_can_access_current_node = node["accessible_from_nodes_hostnames"]
+            addresses = [str(iface.ip) for iface in machine.network_interfaces]
+            accessible_machines = [
+                m.hostname for m in ReportService.get_accessible_machines(machine)
+            ]
             formatted_nodes.append(
                 {
-                    "label": node["label"],
-                    "ip_addresses": node["ip_addresses"],
-                    "accessible_from_nodes": nodes_that_can_access_current_node,
-                    "services": node["services"],
-                    "domain_name": node["domain_name"],
-                    "pba_results": node["pba_results"] if "pba_results" in node else "None",
+                    "label": machine.hostname,
+                    "ip_addresses": addresses,
+                    "accessible_from_nodes": accessible_machines,
+                    "services": [],
+                    "domain_name": "",
+                    "pba_results": "None",
                 }
             )
 
         logger.info("Scanned nodes generated for reporting")
 
         return formatted_nodes
 
-    @staticmethod
-    def get_all_displayed_nodes():
-        nodes_without_monkeys = [
-            NodeService.get_displayed_node_by_id(node["_id"], True)
-            for node in mongo.db.node.find({}, {"_id": 1})
-        ]
-        nodes_with_monkeys = [
-            NodeService.get_displayed_node_by_id(monkey["_id"], True)
-            for monkey in mongo.db.monkey.find({}, {"_id": 1})
-        ]
-        nodes = nodes_without_monkeys + nodes_with_monkeys
-        return nodes
+    @classmethod
+    def get_accessible_machines(cls, machine: Machine):
```

Comment:
accessible != scanned. If a machine A communicated to the island it means that for machine A, the island is accessible. It doesn't mean that machine A scanned the island